src/ui/src/builder/settings/BuilderTemplateInput.vue

```diff
@@ -101,10 +104,10 @@ const props = defineProps({
 	inputId: { type: String, required: false, default: undefined },
 	value: { type: String, required: false, default: undefined },
 	multiline: { type: Boolean, required: false },
-	variant: {
-		type: String as PropType<"code" | "text">,
+	rows: {
+		type: [String, Number] as PropType<number | "half-screen">,
```

Comment:
It's a bit weird that this component can accept `"half-screen"` as the number of rows. It's a sign that you're mixing responsibilities. It should stay generic.

It's better to keep only `Number` here.
```suggestion
		type: Number,
```

Then you need to handle this logic in the parent

https://github.com/writer/writer-framework/blob/c016bf8261d44838d3ac4fe0e8f3fbbe1ce1ca7b/src/ui/src/builder/settings/BuilderFieldsText.vue#L24

like doing 

```
:rows="isExpanded ? undefined : 3"
```

Maybe you'll need to remove with `default: 3` and refactor a bit

Reply:
Thanks Alex, I see your point and it makes total sense to me.

Let me explain my idea behind the change. I used "half-screen" more as a declarative way to control height, similar to how rows works, just more semantic.

If we move that logic to the parent, we'll probably need to expose styles or classes for the inner input/textarea, since right now all styling applies to the root.